influxdb3/write_test.go

```diff
@@ -733,6 +734,42 @@ func TestHttpError(t *testing.T) {
 	assert.ErrorContains(t, err, "error calling")
 }
 
+func TestHttpErrorWithHeaders(t *testing.T) {
+	traceID := "123456789ABCDEF0"
+	tsVersion := "v0.0.1"
+	build := "TestServer"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Trace-Id", traceID)
+		w.Header().Set("X-Influxdb-Build", build)
+		w.Header().Set("X-Influxdb-Version", tsVersion)
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusBadRequest)
+		_, err := w.Write([]byte("{ \"message\": \"Test Response\" }"))
+		if err != nil {
+			assert.FailNow(t, err.Error())
+		}
+	}))
+	defer ts.Close()
+	tc, err := New(ClientConfig{
+		Host:     ts.URL,
+		Token:    "my-token",
+		Database: "my-database",
+	})
+	require.NoError(t, err)
+	err = tc.WriteData(context.Background(), []any{})
+	require.Error(t, err)
+	var serr *ServerError
+	assert.NotPanics(t, func() {
+		errors.As(err, &serr)
+	})
```

Comment:
I think it is safe to assume a correct pointer to `As` is passed since it's declared just before it, I would just make sure it's the correct target type, ie.

```suggestion
        require.True(t, errors.As(err, &serr))
```

Reply:
Changed, however linter wants to enforce this solution: 

```go
require.ErrorAs(t, err, &serr)
```
 So, used the linter suggestion.